### ISSUE TITLE
Support for PW getBy locators

### ISSUE THREAD

**Issue by leeuwe** [MarketSquare/robotframework-browser#2877]
Since PlayWright Version 1.27 there's PlayWrights own take on writing locators, as documented prominently [here](https://playwright.dev/docs/locators). These, you can then [filter](https://playwright.dev/docs/locators#filtering-locators) and more.

These we could implement using something like:
```
Click    pw=getByRole('listitem') -- hasNot: getByText('Product 2')
```
The `--` would be treated a bit like there is custom `>>` for chaining, but `--` would be for filtering.
filtering can consist of: has or hasNot and is followed by either text or [any other locator](https://playwright.dev/docs/locators#filter-by-childdescendant).

**Comment by manykarim** [comment]
While I like the idea to implement the `getBy` Locators and `filters` , I worry that the readability of our Keywords will suffer too much by using the syntax above (e.g. using `--` as an equivalent for `filter`).

One huge advantage of Robot Framework Keywords/Libraries for me was always, that they were easier to read and understand than the equivalent in Playwright, Selenium, etc.

I don't have a good idea how to approach this right now, I just wanted to share my opinion.

**Comment by leeuwe** [comment]
Can you @manykarim elaborate on how this feature will let readibility suffer more than using `>>` as a way of chaining locators?
Locators in general can look terrible, so the way playwright attempted to make an alternative to xpath or css is only to make it more readible I think. Therefore adding the feature in a way that is as close to the original implementation as I could come up with will eventually add to readibility of locators instead.

But writing neat locators will probably always remain a form of art, as writing them badly is so easy

**Comment by Snooz82** [comment]
you can easily find out which selector is similar to what expression by just printing the Locator object in Playwright Python or Playwright Node

Like this:
```python
abs = page.locator("div").filter(has_text=re.compile(r"^ABS 990€$")).get_by_role("checkbox")
print(abs)
```

prints:
```
<Locator frame=<Frame name= url='xxxxxxxx'> selector='div >> internal:has-text=/^ABS 990€$/ >> internal:role=checkbox'>
```

**Comment by aaltat** [comment]
Writing docs is good idea, let’s at least do that. 

Also it would be interesting to see would it be possible write a keyword, for experimenting, that would eat PW locator, like `page.locator("div").filter(has_text=re.compile(r"^ABS 990€$")).get_by_role("checkbox")` Then it would return `<Locator frame=<Frame name= url='xxxxxxxx'> selector='div >> internal:has-text=/^ABS 990€$/ >> internal:role=checkbox'>` but as a valid css locator.

**Comment by Snooz82** [comment]
Will be fixed by #2920

**Comment by Eli-Black-Work** [comment]
Is the `internal:has-text` keyword documented anywhere?

(Just wondering. I've already figured out how to use it 🙂)
